specs/spec_helpers.py

```diff
@@ -1,11 +1,32 @@
 """Defines helpers for specs."""
 
-from flask import Request
+from flask import Request, current_app
+from flask.testing import FlaskClient
+from flask_jwt_extended import create_access_token, get_csrf_token
+from flask_jwt_extended.config import config
 from werkzeug.datastructures import MultiDict, FileStorage
 
 from application import create_app
 
 
+class TestingClient(FlaskClient):
+    """ Adds login to the testing client class """
+
+    def login(self):
+        """ Login a testing user and return the token """
+        access_token = create_access_token(identity='test_user')
+        server_name = current_app.config["SERVER_NAME"] or "localhost"
+        self.set_cookie(server_name,
+                        config.access_cookie_name,
+                        value=access_token,
+                        max_age=config.cookie_max_age,
+                        secure=config.cookie_secure,
+                        httponly=True,
+                        domain=config.cookie_domain,
+                        path=config.access_cookie_path)
+        return get_csrf_token(access_token)
```

Comment:
I don't quite understand what you are doing here. Is this some sort of mock for the login? If yes, why does it have to be mocked, if not what exactly is this? :D

Reply:
I think setting of cookies didn't work properly with the normal login route, that's why I wrote that method
that calls set_cookie of the testing client directly. However, it seems to work with the normal route now, so I will use that instead :D